### PR TITLE
Use typed methods such as setInt, setDouble, setDate ... for prepared statement

### PR DESCRIPTION
**Pull request by tpeyrard**
Use typed methods such as `setInt`, `setDouble`, `setDate` ... for prepared statement.

It moves parsing from the driver to Jmeter, but it allows to compare more easily two drivers / databases. 
Moreover it makes benchmark closer to reality, in fact for performance reasons we try to avoid calls to setObject as the driver as to cast/parse String parameters.

### CODE REVIEW

File: src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java [hunk 1]
```diff
@@ -46,6 +28,14 @@
...
 import org.apache.log.Logger;
 
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.Field;
+import java.sql.*;
+import java.sql.Date;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
```
Comment (milamberspace) [inline src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java]: Please use explicit import, don't use wildcard import.

File: src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java [hunk 2]
```diff
@@ -338,6 +328,44 @@ private static final int countRows(ResultSet resultSet) throws SQLException {
...
     }
 
+    private void setArgument(PreparedStatement pstmt, String argument, int targetSqlType, int index) throws SQLException {
+        switch (targetSqlType) {
+        case Types.INTEGER:
+            pstmt.setInt(index, Integer.parseInt(argument));
+            break;
+        case Types.DOUBLE:
+        case Types.DECIMAL:
+            pstmt.setDouble(index, Double.parseDouble(argument));
```
Comment (vlsi) [inline src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java]: This is not valid.

>https://docs.oracle.com/javase/1.5.0/docs/guide/jdbc/getstart/mapping.html
>The recommended Java mapping for the DECIMAL and NUMERIC types is java.math.BigDecimal

**Comment by Samoht-fr** [comment]
I fixed patch according to your remarks. I also added new conversions according to the [Oracle doc] (https://docs.oracle.com/javase/1.5.0/docs/guide/jdbc/getstart/mapping.html).

File: src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java [hunk 3]
```diff
@@ -18,34 +18,44 @@
...
 
+import org.apache.commons.collections.map.LRUMap;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.save.CSVSaveService;
+import org.apache.jmeter.testelement.AbstractTestElement;
+import org.apache.jmeter.testelement.TestStateListener;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.apache.jmeter.util.JMeterUtils;
+import org.apache.jorphan.logging.LoggingManager;
+import org.apache.log.Logger;
```
Comment (vlsi) [inline src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java]: Please refrain from moving imports around. Can you please move `org.apache` imports back to the initial location?

**Comment by vlsi** [comment]
Looks good except from a glitch with imports moved around.
I'm not sure if additional tests are expected in this PR.

File: src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java (hunk 3 above)
Comment (Samoht-fr) [inline src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java]: That's my IntelliJ ... I'll fix it :)

**Comment by milamberspace** [comment]
Thanks. In the trunk.

URL: http://svn.apache.org/r1700060
Log:
Use typed methods such as setInt, setDouble, setDate ... for prepared statement #27

Bugzilla Id: 58301

This closes #27


Modified:
    jmeter/trunk/src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java
    jmeter/trunk/xdocs/changes.xml

**Comment by Samoht-fr** [comment]
Thanks too.

**Pull request closed without merging**